figures/S8.supplemental_figure8/scripts/S8.r

```diff
@@ -0,0 +1,133 @@
+# get the rows that contain the F1 scores
+f1_scores <- training_metrics[training_metrics$metric == "f1-score",]
+# remove the rows that contain the macro and weighted averages
+f1_scores <- f1_scores[!grepl("macro avg", f1_scores$label),]
+f1_scores <- f1_scores[!grepl("weighted avg", f1_scores$label),]
+# muatate the label column for multiple cases
+f1_scores$label <- gsub("healthy", "Control", f1_scores$label)
+f1_scores$label <- gsub("apoptosis", "Apoptosis", f1_scores$label)
+f1_scores$label <- gsub("pyroptosis", "Pyroptosis", f1_scores$label)
+# mutate the data type column
+f1_scores$group <- gsub("train", "Training", f1_scores$group)
+f1_scores$group <- gsub("test", "Testing", f1_scores$group)
+f1_scores$group <- gsub("validation", "Validation", f1_scores$group)
+f1_scores$group <- gsub("treatment_holdout", "Treatment Holdout", f1_scores$group)
+f1_scores$group <- gsub("holdout", "Holdout", f1_scores$group)
+# factorize the group column
+f1_scores$group <- factor(f1_scores$group, levels = c(
+    "Training", "Validation", "Testing","Treatment Holdout", "Holdout"
+))
+# mutate the shuffled_data column
+f1_scores$shuffle <- gsub("TRUE", "Shuffled", f1_scores$shuffle)
+f1_scores$shuffle <- gsub("FALSE", "Not Shuffled", f1_scores$shuffle)
+# cbind the support column to the f1_scores df
+f1_scores <- cbind(f1_scores, support$value)
+# rename the support column
+colnames(f1_scores)[colnames(f1_scores) == "support$value"] <- "support"
+# dived the support by 10,000 to get the number of cells
+f1_scores$support <- f1_scores$support / 10000
+# round the support column to 2 decimal places
+f1_scores$support <- round(f1_scores$support, 2)
+
+
+# make the label a factor so that the order is preserved
+f1_scores$label <- factor(
+    f1_scores$label, levels = c(
+        "Control", "Apoptosis", "Pyroptosis"
+        )
+    )
+
+
+head(f1_scores, 1)
+
+# remove the treatment holdout rows
+f1_scores <- f1_scores[!grepl("Treatment Holdout", f1_scores$group),]
+
+# set plot size
+width <- 10
+height <- 5
+options(repr.plot.width = width, repr.plot.height = height)
+# bar plot of the F1 scores
+f1_score_plot <- (
+    ggplot(f1_scores, aes(x = shuffle, y = value, fill = group))
+    + geom_bar(stat = "identity", position = "dodge")
+
+    + ylim(0, 1)
+    + facet_wrap(~label)
+    + ylab("F1 Score")
+    + xlab("Data Split")
+    # change the legend title
+    + labs(fill = "Predicted Class")
+    # change the colours
+    + scale_fill_manual(values = c(
+        "Training" = "#88F2F2",
+        "Validation" = "#056CF2",
+        "Testing" = "#A6382E",
+        "Holdout" = "#F2A900"
+    ))
+    + figure_theme_wide
+
+)
+ggsave(f1_plot_path, f1_score_plot, width = width, height = height, dpi = 600)
+f1_score_plot
+
+
+
+# set plot size
+width <- 10
+height <- 10
+options(repr.plot.width=width, repr.plot.height=height, units = "cm", dpi = 600)
+fig5 <- (
+    f1_score_plot
+    + plot_annotation(tag_levels = 'A') & theme(plot.tag = element_text(size = 20))
```

Comment:
Just curious, is there a reason you include "panel A" if this figure is stand alone or are you planning on making S8 also include other plots? I have seen your other figures with only one plot not include a panel letter.

Reply:
I need to remove that